src/hw25/LessonDao.java

```diff
@@ -0,0 +1,80 @@
+package hw25;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LessonDao {
+        private Connection connection;
+
+        public LessonDao(Connection connection) {
+            this.connection = connection;
+        }
+
+        public void addLesson(Lesson lesson) throws SQLException {
+            String query = "INSERT INTO Lesson (name, homework_id) VALUES (?, ?)";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, lesson.getName());
+            statement.setInt(2, lesson.getHomework().getId());
+            statement.executeUpdate();
+            statement.close();
+        }
+
+        public void deleteLesson(int lessonId) throws SQLException {
+            String query = "DELETE FROM Lesson WHERE id = ?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setInt(1, lessonId);
+            statement.executeUpdate();
+            statement.close();
+        }
+
+        public List<Lesson> getAllLessons() throws SQLException {
+            List<Lesson> lessons = new ArrayList<>();
+            String query = "SELECT l.id, l.name, h.id, h.name, h.description " +
+                    "FROM Lesson l INNER JOIN Homework h ON l.homework_id = h.id";
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                int lessonId = resultSet.getInt(1);
```

Comment:
1,2,3,4,5 are magic numbers - it is unclear what they are responsible for. Create a local variable `int index = 1;`, and increment it each time when passing it as a parameter to the result set. When retrieving the last parameter, an increment is not needed.